float32[4] dstate
